File: SampleProject/SampleProject/SecondViewController.m
//
//  SecondViewController.m
//  SampleProject
//
//  Created by Pollfish Inc. on 11/3/15.
//  Copyright © 2015 POLLFISH. All rights reserved.
//

#import "SecondViewController.h"

@interface SecondViewController ()

@end

@implementation SecondViewController

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    if (@available(iOS 14, *)) {
        [self requestIDFAPermission];
    } else {
        [self initPollfish];
    }

}

- (void)requestIDFAPermission {
#if __has_include(<AppTrackingTransparency/AppTrackingTransparency.h>)
    [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
      dispatch_async(dispatch_get_main_queue(), ^{
          if (status == ATTrackingManagerAuthorizationStatusAuthorized) {
              [self initPollfish];
          } else {
              [self showNoPermissionAlert];
          }
      });
    }];
#endif
}

- (void) initPollfish{
        
    NSLog(@"Pollfish: init");
    
    PollfishParams *pollfishParams = [[PollfishParams alloc] init:@"YOUR_API_KEY"];
    [pollfishParams releaseMode:false];
    [pollfishParams offerwallMode:false];
    [pollfishParams rewardMode:true];
    [pollfishParams requestUUID:@"my_id"];
    
    [Pollfish initWith:pollfishParams delegate:self];
    
    _incentivizeBtn.hidden=true;
    _loggingLabel.text=@"Logging area..";
}

- (void)pollfishSurveyReceivedWithSurveyInfo:(SurveyInfo *)surveyInfo
{
    
    int surveyPrice = [[surveyInfo cpa] intValue];
    int surveyIR = [[surveyInfo ir] intValue];
    int surveyLOI = [[surveyInfo loi] intValue];
    
    NSString *surveyClass = [surveyInfo surveyClass];
    
    NSString *rewardName = [surveyInfo rewardName];
    int rewardValue = [[surveyInfo rewardValue] intValue];
    
    NSLog(@"Pollfish: Survey Completed - SurveyPrice:%d andSurveyIR: %d andSurveyLOI:%d andSurveyClass:%@ andRewardName:%@ andRewardValue:%d", surveyPrice,surveyIR, surveyLOI, surveyClass, rewardName, rewardValue);
    
    [_incentivizeBtn setTitle:[NSString stringWithFormat:@"Win %d %@ by completing a survey",rewardValue, rewardName] forState:UIControlStateNormal];

    _incentivizeBtn.hidden=false;
    
    _loggingLabel.text=[NSString stringWithFormat:@"Pollfish: Survey Received - SurveyPrice:%d andSurveyIR: %d andSurveyLOI:%d andSurveyClass:%@ andRewardName:%@ andRewardValue:%d", surveyPrice,surveyIR, surveyLOI, surveyClass, rewardName, rewardValue];
}

- (void)pollfishSurveyCompletedWithSurveyInfo:(SurveyInfo *)surveyInfo
{
    int surveyPrice = [[surveyInfo cpa] intValue];
    int surveyIR = [[surveyInfo ir] intValue];
    int surveyLOI = [[surveyInfo loi] intValue];
    
    NSString *surveyClass = [surveyInfo surveyClass];
    
    NSString *rewardName = [surveyInfo rewardName];
    int rewardValue = [[surveyInfo rewardValue] intValue];
    
    NSLog(@"Pollfish: Survey Completed - SurveyPrice:%d andSurveyIR: %d andSurveyLOI:%d andSurveyClass:%@ andRewardName:%@ andRewardValue:%d", surveyPrice,surveyIR, surveyLOI, surveyClass, rewardName, rewardValue);
    
    _loggingLabel.text=[NSString stringWithFormat:@"Pollfish: Survey Completed - Congratulations, you have won %d %@", rewardValue, rewardName];
    
    // in a real world app you should wait for s2s callbacks prior rewarding your user
    
    _incentivizeBtn.hidden=true;
}

- (void)pollfishOpened
{
    NSLog(@"Pollfish: Survey Opened");
    
    _loggingLabel.text=@"Pollfish: Survey Opened";
}

- (void)pollfishClosed
{
    NSLog(@"Pollfish: Survey Closed");
}

- (void)pollfishNotAvailable
{
    NSLog(@"Pollfish: Survey Not Available");
    
    _loggingLabel.text=@"Pollfish: Survey Not Available";
    _incentivizeBtn.hidden=true;
}

- (void)pollfishUsernotEligible
{
    NSLog(@"Pollfish: User Not Eligible");
    
    _loggingLabel.text=@"Pollfish: User Not Eligible";
    _incentivizeBtn.hidden=true;
}

-(IBAction)showPollfish:(id)sender{
    [Pollfish show];
}

- (void) pollfishUserRejectedSurvey
{
    NSLog(@"Pollfish: User Rejected Survey");
    
    _loggingLabel.text=@"Pollfish: User Rejected Survey";
    _incentivizeBtn.hidden=true;
}

@end

File: SampleProjectSwift/SampleProjectSwift/SecondViewController.swift
//
//  SecondViewController.swift
//  SampleProjectSwift
//
//  Created by Pollfish Inc. on 11/4/15.
//  Copyright © 2015 POLLFISH. All rights reserved.
//

import UIKit
import Pollfish
#if canImport(AppTrackingTransparency)
import AppTrackingTransparency
#endif

class SecondViewController: UIViewController, PollfishDelegate {

    @IBOutlet weak var incentivizeBtn: UIButton!
    @IBOutlet weak var loggingLabel: UILabel!
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        loggingLabel.text="Logging area.."
        incentivizeBtn.isHidden=true
        
        if #available(iOS 14, *) {
            requestIDFAPermission()
        } else {
            pollfishInit()
        }
        
        loggingLabel.text="Logging area.."
    }
    
    @available(iOS 14, *)
    func requestIDFAPermission() {
        #if canImport(AppTrackingTransparency)
        ATTrackingManager.requestTrackingAuthorization { status in
            DispatchQueue.main.async {
                switch status {
                case .authorized:
                    self.pollfishInit()
                default:
                    self.showNoPermissionAlert()
                }
            }
        }
        #endif
    }
    
    func pollfishInit(){
        let pollfishParams = PollfishParams("2ae349ab-30b8-4100-bc4d-b33b82e76519")
            .rewardMode(true)
            .releaseMode(false)
            .requestUUID("my_id")
                
        Pollfish.initWith(pollfishParams, delegate: self)
    }
        
    func pollfishSurveyNotAvailable() {
        print("pollfishNotAvailable")
        
        loggingLabel.text="Pollfish - Survey Not Available"
    }
    
    func pollfishSurveyReceived(surveyInfo: SurveyInfo?) {
        let surveyPrice = surveyInfo?.cpa?.intValue
        let surveyIR = surveyInfo?.ir?.intValue
        let surveyLOI = surveyInfo?.loi?.intValue
        let surveyClass = surveyInfo?.surveyClass
        
        let rewardName = surveyInfo?.rewardName
        let rewardValue = surveyInfo?.rewardValue?.intValue
        
        print("Pollfish Survey  Received - SurveyPrice: \(String(describing: surveyPrice)) andSurveyIR: \(String(describing: surveyIR)) andSurveyLOI: \(String(describing: surveyLOI)) andSurveyClass: \(String(describing: surveyClass)) andRewardName: \(String(describing: rewardName)) andRewardValue:\(String(describing: rewardValue))")
        
        loggingLabel.text="Pollfish Survey Received - SurveyPrice: \(surveyPrice ?? 0) andSurveyIR: \(surveyIR ?? 0) andSurveyLOI: \(surveyLOI ?? 0) andSurveyClass: \(surveyClass ?? "") andRewardName: \(rewardName ?? "") andRewardValue:\(rewardValue ?? 0)"
        
        incentivizeBtn.setTitle("Earn \(rewardValue ?? 0) \(rewardName ?? "") by completing a survey"
            , for: UIControl.State.normal)
        
        incentivizeBtn.isHidden=false
    }
    
    func pollfishOpened() {
        print("pollfishOpened")
        
        loggingLabel.text="Pollfish - Pollfish Panel Opened"
    }
    
    func pollfishClosed() {
        print("pollfishClosed")
    }
    
    func pollfishUserNotEligible() {
        print("pollfishUsernotEligible")
        
        loggingLabel.text="Pollfish - User Not Eligible"
        incentivizeBtn.isHidden=true
    }
    
    func pollfishUserRejectedSurvey() {
        print("pollfishUserRejectedSurvey")
        
        loggingLabel.text="Pollfish - User Rejected Survey"
        incentivizeBtn.isHidden=true
    }
    
    func pollfishSurveyCompleted(surveyInfo: SurveyInfo) {
        let surveyPrice = surveyInfo.cpa?.intValue
        let surveyIR = surveyInfo.ir?.intValue
        let surveyLOI = surveyInfo.loi?.intValue
        let surveyClass = surveyInfo.surveyClass
        
        let rewardName = surveyInfo.rewardName
        let rewardValue = surveyInfo.rewardValue?.intValue
        
        print("Pollfish Survey  Completed - SurveyPrice: \(String(describing: surveyPrice)) andSurveyIR: \(String(describing: surveyIR)) andSurveyLOI: \(String(describing: surveyLOI)) andSurveyClass: \(String(describing: surveyClass)) andRewardName: \(String(describing: rewardName)) andRewardValue:\(String(describing: rewardValue))")
        
        loggingLabel.text="Pollfish Survey Completed - You won \(rewardValue ?? 0) \(rewardName ?? "")"
        
        // in a real world app you should wait for s2s callbacks prior rewarding your user
        
        incentivizeBtn.isHidden=true
    }
    
    @IBAction func showPollfish(_ sender: AnyObject) {
        print("showPollfish")
        
        Pollfish.show();
    }
    
    @IBAction func hidePollfish(_ sender: AnyObject) {
        print("hidePollfish")
        
        Pollfish.hide();
    }
    
}

